props/saber_BC_buttons.h

```diff
@@ -477,738 +1492,1266 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
+      speaking_ = true;
+      SaberBase::DoEffect(EFFECT_BATTERY_LEVEL, 0);
+      return;
+    } else {
+      sound_library_.SayNumber(battery_monitor.battery_percent(), SAY_WHOLE);
+      sound_library_.SayPercent();
+      PVLOG_NORMAL << "Battery Percentage: " <<battery_monitor.battery_percent() << "%\n";
+      speaking_ = true;
+      SaberBase::DoEffect(EFFECT_BATTERY_LEVEL, 0);
+    }
+    return;
+  }
 
-#ifdef BC_TWIST_ON
-    case EVENTID(BUTTON_NONE, EVENT_TWIST, MODE_OFF):
-      if (mode_volume_ || scroll_presets_) return false;
-#ifdef NO_BLADE_NO_GEST_ONOFF
-      if (!blade_detected_) return false;
-#endif
-      // Delay twist events to prevent false trigger from over twisting
-      if (millis() - last_twist_ > 2000 &&
-        millis() - saber_off_time_ > 1000) {
-        FastOn();
-#ifdef BC_GESTURE_AUTO_BATTLE_MODE
-        PVLOG_NORMAL << "Entering Battle Mode\n";
-        battle_mode_ = true;
-#endif
-        last_twist_ = millis();
-      }
-      return true;
-#endif  // BC_TWIST_ON
+  void OnDemandBatteryLevel() {
+    if (scroll_presets_) return;
+      PVLOG_NORMAL << "Battery Voltage: " << battery_monitor.battery() << "\n";
+      PVLOG_NORMAL << "Battery Percentage: " <<battery_monitor.battery_percent() << "\n";
+      SaberBase::DoEffect(EFFECT_BATTERY_LEVEL, 0);
+      return;
+  }
 
-#ifdef BC_TWIST_OFF
-    case EVENTID(BUTTON_NONE, EVENT_TWIST, MODE_ON):
-#ifdef NO_BLADE_NO_GEST_ONOFF
-      if (!blade_detected_) return false;
-#endif
-      // Delay twist events to prevent false trigger from over twisting
-      if (millis() - last_twist_ > 3000) {
-        last_twist_ = millis();
-        TurnOffHelper();
-      }
-      return true;
-#endif  // BC_TWIST_OFF
+  void DoTrackStartOrStop() {
+    if (scroll_presets_ || spam_blast_) return;
+    PVLOG_NORMAL << "** Track playback Started or Stopped\n";
+    StartOrStopTrack();
+  }
 
-#ifdef BC_STAB_ON
-    case EVENTID(BUTTON_NONE, EVENT_STAB, MODE_OFF):
-      if (mode_volume_) return false;
-#ifdef NO_BLADE_NO_GEST_ONOFF
-      if (!blade_detected_) return false;
-#endif
-      // Delay Stab On at boot
-      if (millis() - saber_off_time_ > 1000) {
-        FastOn();
-#ifdef BC_GESTURE_AUTO_BATTLE_MODE
-        PVLOG_NORMAL << "Entering Battle Mode\n";
-        battle_mode_ = true;
-#endif
-      }
-      return true;
-#endif  // BC_STAB_ON
+  void DoLockup() {
+    if (!SaberBase::Lockup() && SaberBase::IsOn()) {
 
-#ifdef BC_THRUST_ON
-    case EVENTID(BUTTON_NONE, EVENT_THRUST, MODE_OFF):
-      if (mode_volume_) return false;
-#ifdef NO_BLADE_NO_GEST_ONOFF
-      if (!blade_detected_) return false;
-#endif
-      // Delay Thrust On at boot
-      if (millis() - saber_off_time_ > 1000) {
-        FastOn();
-#ifdef BC_GESTURE_AUTO_BATTLE_MODE
-        PVLOG_NORMAL << "Entering Battle Mode\n";
-        battle_mode_ = true;
-#endif
+  #ifdef BC_DUAL_BLADES
+      if (isMainBladeOn() && !isSecondBladeOn()) {
```

Comment:
weird indentation

Reply:
The preprocessor lines weren't left justified, others look ok though, no?